MetaMask/metamask-extension
# Chore/refactor smart swap modal

## Explanation



This PR resolves [MMS-520](https://consensyssoftware.atlassian.net/browse/MMS-520):

1. Updates the Smart Transaction popover to use the new components from the Component Library and values from the Design System. 
2. The component was converted to TS for better typing. 
3. A Storybook story was added for ease of testing and development.

## Screenshots/Screencaps



### Before


![image-20230727-124737](https://github.com/MetaMask/metamask-extension/assets/139582705/4083508e-ed69-4479-a5be-f80e7737629e)

### After


![Screenshot 2023-07-27 at 4 10 47 PM](https://github.com/MetaMask/metamask-extension/assets/139582705/2dcb10dc-18b0-42da-9cba-45b08c0b2552)


## Manual Testing Steps



1. Wipe your MM data by reinstalling the extension
4. Navigate to the Swaps screen
5. Observe modal popping up
6. Accept or reject the Smart Tx 
7. Click the Settings in Swaps
8. Observe that your Accept/Reject was properly set for Smart Tx

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
